src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,35 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import classNames from 'classnames';
+
+interface Props {
+  id: number;
+  title: string;
+  completed: boolean;
+  name: string;
+  setTargetTodo: (todo: null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  id,
+  title,
+  completed,
+  name,
+  setTargetTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setInterval(() => {
+      setIsLoading(false);
+    }, 300);
```

Comment:
Using `setInterval` to manage loading state is not ideal. Consider using a more reliable method, such as a timeout with `setTimeout`, to change the loading state after the data has been fetched.